nativelib/src/main/resources/scala-native/platform/posix/process_monitor.cpp

```diff
@@ -7,13 +7,21 @@
 #include <sys/wait.h>
 #include <sys/time.h>
 #include <unordered_map>
-#ifdef __APPLE__
-// Semaphores on OSX are deprectated
-#include <dispatch/dispatch.h>
-#else
 #include <semaphore.h>
 #include <errno.h>
 #include <sys/mman.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+
+// The +1 accounts for the null char at the end of the name
+#ifdef __APPLE__
+#include <sys/posix_sem.h>
+#define SEM_MAX_LENGTH PSEMNAMLEN + 1
+#else
+#include <limits.h>
+#define SEM_MAX_LENGTH _POSIX_PATH_MAX + 1
```

Comment:
This is and creation of semaphore is copy-pasted from Phase.c in Commix impl. In both cases this is needed due to problems on MacOs - usage of sem_create would return -1 and errno would mention not implemented error. 
Also because previous `dispatch_semaphore_t` was leading to deadlocks on Arm64 we do replace both linux and mac implementation with shared named semaphore